### ISSUE TITLE
Umlauts in "display name" made account unuseable

### ISSUE THREAD

**Issue by controlcde** [fruux/sabre-dav#791]
I´ve add users to the database with Umlauts in their display names: And I get error on using calendar and addressbook on OSX. After changing the the Umlauts to normal characters the access works. The  error messages are needed?

**Comment by evert** [comment]
Umlauts should definitely work. My first guess is that you are not using the correct encoding. Everything in the database must be UTF-8 for things to work.

**Comment by staabm** [comment]
@evert yesterday I learned that mac has a different utf8 then others, namely NFD vs NFC


NFC = Normalization Form Canonical Composition
NFD = Normalization Form Canonical Decomposition

**Comment by evert** [comment]
What you're talking about only really affects filesystems as far as I know, and even then it actually doesn't cause issues. Both are UTF-8 and fully understood by sabre/dav. In this particular case the user is talking about the displayname. If it were the `uri` field then I agree that it could be problematic for some clients.

**Comment by controlcde** [comment]
SabreDAV runs on Debian Jessie and I execute a script to add the user. The command line is
```
mysql -u root -p sabredav <<EOF
INSERT INTO users (username, digesta1) VALUES ('testuser1',MD5('testuser1:SabreDAV:password'));
INSERT INTO principals (uri, email, displayname) VALUES ('principals/testuser1','testuser1@domain.de','Ein Benützer');
EOF
```

After adding this account to iCal I get the error: `Die Anfrage (CalDAVAccountRefreshQueueableOperation) für „Account „testcal.controlc.de““ ist fehlgeschlagen.`

**Comment by evert** [comment]
Can you run:

    mysql -u root -p sabredav -e "SELECT displayname FROM principals WHERE uri = 'principals/testuser1';" | hexdump -C

And send the output here?

**Comment by controlcde** [comment]
```
00000000  64 69 73 70 6c 61 79 6e  61 6d 65 0a 45 69 6e 20  |displayname.Ein |
00000010  42 65 6e c3 bc 74 7a 65  72 0a                    |Ben..tzer.|
0000001a
```

**Comment by evert** [comment]
Unfortunately the umlaut is encoded as c3 bc, which is what we'd expect for UTF-8. Could you try changing your connecting string (DSN) from:

    $pdo = new PDO('...')

to include:

    ;charset=utf8

At the end? Curious if that fixes things.

**Comment by Serverfrog** [comment]
@evert Thank you. 
I had a issue that Umlauts where not Printed in the HTML View Version, so i used rawurlencode. But this broke the Microsoft-WebDAV-MiniRedir client. After i used the Charset Parameter everything worked.

TL;DR @controlcde @evert The Charset Parameter Worked

**Comment by staabm** [comment]
@evert maybe we should check whether the used db connection uses utf8 charset?
http://php.net/manual/en/mysqli.get-charset.php

whether this charset is configured on db side or per connection doesnt make a difference for sabre-dav, so checking for it based on the link should work...

**Comment by evert** [comment]
Agreed. This must be added to all the examples everywhere.

**Comment by staabm** [comment]
maybe we can even add it to the boot sequence of the setup.

**Comment by christophdb** [comment]
Hi,

I can confirm that this bug also is relevant for emclient.
As soon as in the table _principals_ the field _displayname_ has an Umlaut like "äüö" or "ß", it is not possible to synchronize with emclient. Even if the user has only read rights to the calenders of a user with an Umlaut the synchronisation stops working.

emClient is showing the following error message:
```
17:59:03 franck [CalDAV / CardDAV]  MailClient.Accounts.ConnectionException: Ungültiges Zeichen in der angegebenen Codierung. Zeile 13, Position 31. ---> System.Xml.XmlException: Ungültiges Zeichen in der angegebenen Codierung. Zeile 13, Position 31.
17:59:03    bei System.Xml.XmlTextReaderImpl.Throw(Exception e)
17:59:03    bei System.Xml.XmlTextReaderImpl.Throw(String res, String arg)
17:59:03    bei System.Xml.XmlTextReaderImpl.InvalidCharRecovery(Int32& bytesCount, Int32& charsCount)
...
```

I tried to force a charcode switch with the following mysql orders:
ALTER TABLE principals CHARACTER SET utf8 COLLATE utf8_general_ci;
ALTER TABLE principals CONVERT TO CHARACTER SET utf8 COLLATE utf8_general_ci;
but it did not help.

In the end I changed all displayname's from _Müller_ to _Mueller_.
I will do some research within the next days and will post my learnings in here.

Best regards
Christoph
